mesheryctl/internal/cli/root/system/config.go

```diff
@@ -85,6 +100,95 @@ func getContexts(configFile, tokenPath string) ([]string, error) {
 	return contexts, nil
 }
 
+// Tries to get AWS user credentials required for out of cluster deployments
+// to work.
+// Firstly looks for them in environment variables AWS_ACCESS_KEY_ID
+// and AWS_SECRET_ACCESS_KEY
+// Secondly in the shared directory at $HOME/.aws/
+// Lastly tries to take user input and generates a file in ~/.meshery/aws
+// TODO: Support EC2 instance role credentials
+// https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.40.43#readme-configuring-credentials
+func awsCredentialHelper() error {
+	log.Info("Looking for aws user credentials in $AWS_ACCESS_KEY_ID and $AWS_SECRET_ACCESS_KEY OR ~/.aws/credentials...")
+	sess := session.Must(session.NewSession())
+	creds := sess.Config.Credentials
+	credValue, err := creds.Get()
+
+	if err != nil {
+		log.Info("Credentials not found. Generating static credentials now...")
+
+		var accessKeyID, secretAccessKey string
+		log.Println("Enter `aws_access_key_id`: ")
+		_, err := fmt.Scanf("%s", &accessKeyID)
+		if err != nil {
+			// err reading entered cred
+			return err
+		}
+		log.Println("Enter `aws_secret_access_key`: ")
+		_, err = fmt.Scanf("%s", &secretAccessKey)
+		if err != nil {
+			// err reading entered cred
+			return err
+		}
+
+		creds = credentials.NewStaticCredentials(accessKeyID, secretAccessKey, "")
+		credValue, err = creds.Get()
+		if err != nil {
+			// entered creds invalide
+			return err
+		}
+	}
+
+	// since there is no way yet to check expired creds for StaticProvider
+	// we only check creds obtained from shared directories or environment
+	// variables
+	if credValue.ProviderName == envConfigProvider || credValue.ProviderName == sharedConfigProvider {
+		if creds.IsExpired() {
+			// credentials have expired. generate new
+			return err
+		}
+	}
+
+	// write the tokens to ~/.aws/credentials in case of static and env credentials
+	// providers so that it can be copied into the docker container
+	if credValue.ProviderName == envConfigProvider || credValue.ProviderName == staticProvider {
+		err = createSharedProviderFile(credValue)
+		if err != nil {
+			return err
+		}
+		err = utils.DockerCp("/meshery_meshery_1", mesheryAWS, "/home/appuser/.aws")
```

Comment:
Will a non-root user executing mesheryctl have access to copy files into the meshery docker image?

Reply:
Hmm... only if they're in the `docker` user group which is a basic requirement to use docker :thinking: